connect/models/asset.py

```diff
@@ -15,7 +15,17 @@
 
 
 class Asset(BaseModel):
-    pass
+    def get_parameter_by_id(self, id_):
+        try:
+            return [a for a in self.params if a.id == id_][0]
+        except IndexError:
+            return None
+
+    def get_item_by_id(self, id_):
+        try:
+            return [a for a in self.items if a.id == id_][0]
+        except IndexError:
+            return None
 
```

Comment:
Why not?
```
def _get_by_id(objects, identity):
    try:
        return [obj for obj in objects if obj.id == identity][0]
    except IndexError:
        return None

def get_parameter_by_id(self, identity):
    return self._get_by_id(self.params, identity)

def get_item_by_id(self, identity):
    return self._get_by_id(self.items, identity)
```

Reply:
And move this method `_get_by_id` to BaseModel

Reply:
Thank you, it's much better like that. I have also used filter() in _get_by_id as you suggested for the Fulfillment methods.

Reply:
But you no wrote tests :(

Reply:
I'm on it! I'll update the PR soon :)